scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUri.scala

```diff
@@ -109,15 +111,39 @@ private[sparkey] class SparkeyWriter(val uri: SparkeyUri) {
       case gcsUri: GcsSparkeyUri => {
         // Copy .spi and .spl to GCS path
         for (ext <- Seq("spi", "spl")) {
-          val writer = gcsUri.gcs.create(
-            GcsPath.fromUri(s"${gcsUri.basePath}.$ext"),
-            "application/octet-stream")
-          writer.write(ByteBuffer.wrap(Files.readAllBytes(Paths.get(s"$localFile.$ext"))))
-          writer.close()
+          val localFileSource = new File(s"$localFile.$ext")
+          val gcsPath = GcsPath.fromUri(s"${gcsUri.basePath}.$ext")
+          bufferedCopyLocalToGcs(gcsUri.gcs, localFileSource, gcsPath)
         }
       }
       case _ => ()
     }
   }
 
+
+  private def bufferedCopyLocalToGcs(gcsUtil: GcsUtil, localFileSource: File, gcsPath: GcsPath): Unit = {
```

Comment:
But anyway - why implement this? Why not use channels (even zero copy), like for example (pseudo code):

```
val input = new FileInputStream(localFileSource)
val channel = input.getChannel
channel.transferTo(0, channel.size(), gcsFile)
```